File: javascript/1437.Check-If-All-1's-Are-at-Least-Length-K-Places-Away.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var kLengthApart = function (nums, k) {
  let length = k;

  for (const num of nums) {
    if (num === 1) {
      if (length < k) {
        return false;
      } else {
        length = 0;
      }
    } else {
      length += 1;
    }
  }

  return true;
};

File: python/1437.Check-If-All-1's-Are-at-Least-Length-K-Places-Away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        length = k

        for num in nums:
            if num == 1:
                if length < k:
                    return False
                else:
                    length = 0
            else:
                length += 1

        return True